### ISSUE TITLE
legends on graph when graphic two different geom_smooths on one graph

### ISSUE THREAD

**Issue by thekateblock** [STAT545-UBC/Discussion#352]
I have the following code that creates an weighted mean by year. Now I would like to graph 2 lines onto 1 single graph. One is the weighted and one is the unweighted mean by year. I have managed to make the lines appear and look different. BUT: I have read a bunch of documentation, but i cannot figure out how to make a legend appear that tells the viewer which one is which.

Here is my code:

new_gap <- my_gap %>% group_by(year) %>% summarise (lifeExp_weightedM = weighted.mean(lifeExp,pop), M_unweighted = mean(lifeExp))

new_gap %>% ggplot() + 
  geom_smooth(color= "hotpink", aes(year, lifeExp_weightedM)) + 
  geom_smooth(color = "purple", linetype= 2, aes(year,M_unweighted))

**Comment by jennybc** [comment]
I don't know that off the top of my head. When I googled "ggplot2 custom legend" the top hits looked pretty promising, so that might be a fruitful search phrase.

**Comment by samhinshaw** [comment]
So this leads into some interesting discussion on ggplot2 behavior and "tidy" data.  By default, ggplot2 will only create a legend when you have specified a dependent variable within `aes()`.  The reason you're not seeing a legend is because you have manually specified line colors here.  We'll learn more about this next week, but here's a sneak preview:
```
library(tidyr)
tidy_gap <- tidyr::gather(new_gap, "M_type", "M_lifeExp", 2:3)
tidy_gap %>% arrange(year) %>% head(n = 4)
```

| year|M_type            | M_lifeExp|
|----:|:-----------------|---------:|
| 1952|lifeExp_weightedM |  48.94424|
| 1952|M_unweighted      |  49.05762|
| 1957|lifeExp_weightedM |  52.12189|
| 1957|M_unweighted      |  51.50740|

```
ggplot(tidy_gap, aes(x = year, y = M_lifeExp, color = M_type)) + 
  geom_smooth()
```
![image](https://cloud.githubusercontent.com/assets/14186005/19084828/e6a4d402-8a1c-11e6-8de8-cca5020112f9.png)

**Comment by thekateblock** [comment]
Thanks so much @samhinshaw

**Comment by thekateblock** [comment]
@samhinshaw can you explain what the "2:3" does in the gather?

**Comment by samhinshaw** [comment]
Yes! `2:3` is a lazy way to select the columns `lifeExp_weightedM` and `M_unweighted` in the `new_gap` data.frame.  You could similarly use
`tidy_gap <- gather(new_gap, "M_type", "M_lifeExp", c(lifeExp_weightedM, M_unweighted))`

Huh, looks like you guys went over this today in class, I was still going off of the older syllabus 😆. I highly recommend Garrett Grolemund's primers on tidyr ([here is one](http://garrettgman.github.io/tidying/)), as well as the RStudio [tidyr cheatsheet](https://www.rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf).  And of course, Jenny's [Lord of the Rings example](https://github.com/jennybc/lotr-tidy). 

Getting familiar with "wide" vs "long" data formats will help you tremendously with data analysis and plotting with ggplot2 😄

**Comment by jennybc** [comment]
> @samhinshaw can you explain what the "2:3" does in the gather?

Great example of why it's better to use variable names than column numbers. 😉

**Comment by sinaneza** [comment]
@samhinshaw 
so helpful
